monai/deploy/operators/clara_viz_operator.py

```diff
@@ -0,0 +1,133 @@
+@md.input("seg_image", Image, IOType.IN_MEMORY)
+@md.env(pip_packages=["clara.viz.core", "clara.viz.widgets", "IPython"])
+class ClaraVizOperator(Operator):
+    """
+    This operator uses Clara Viz to provide interactive view of a 3D volume including segmentation mask.
+    """
+
+    def __init__(self):
+        """Constructor of the operator."""
+        super().__init__()
+
+    @staticmethod
+    def _build_array(image, order):
+        if order == "MXYZ":
+            # @todo mask is transposed, see issue #171
+            numpy_array = np.transpose(image.asnumpy(), (2, 1, 0))
+        else:
+            numpy_array = image.asnumpy()
+
+        array = DataDefinition.Array(array=numpy_array, order=order)
+        array.element_size = [1.0]
+        array.element_size.append(image.metadata().get("col_pixel_spacing", 1.0))
+        array.element_size.append(image.metadata().get("row_pixel_spacing", 1.0))
+        array.element_size.append(image.metadata().get("depth_pixel_spacing", 1.0))
+
+        # the renderer is expecting data in RIP order (Right Inferior Posterior) which results in
+        # this matrix
+        target_affine_transform = [
+            [0.0, 1.0, 0.0, 0.0],
+            [1.0, 0.0, 0.0, 0.0],
+            [0.0, 0.0, -1.0, 0.0],
+            [0.0, 0.0, 0.0, 1.0],
+        ]
+
+        dicom_affine_transform = image.metadata().get("dicom_affine_transform", np.identity(4))
+
+        affine_transform = np.matmul(target_affine_transform, dicom_affine_transform)
+
+        array.permute_axes = [
+            0,
+            max(range(3), key=lambda k: abs(affine_transform[0][k])) + 1,
+            max(range(3), key=lambda k: abs(affine_transform[1][k])) + 1,
+            max(range(3), key=lambda k: abs(affine_transform[2][k])) + 1,
+        ]
+
+        array.flip_axes = [
+            False,
+            affine_transform[0][array.permute_axes[1] - 1] < 0.0,
+            affine_transform[1][array.permute_axes[2] - 1] < 0.0,
+            affine_transform[2][array.permute_axes[3] - 1] < 0.0,
+        ]
+
+        return array
+
+    def compute(self, op_input: InputContext, op_output: OutputContext, context: ExecutionContext):
+        """Displays the input image and segmentation mask
+
+        Args:
+            op_input (InputContext): An input context for the operator.
+            op_output (OutputContext): An output context for the operator.
+            context (ExecutionContext): An execution context for the operator.
+        """
+        input_image = op_input.get("image")
+        if not input_image:
+            raise ValueError("Input image is not found.")
+        input_seg_image = op_input.get("seg_image")
+        if not input_seg_image:
+            raise ValueError("Input segmentation image is not found.")
+
+        # build the data definition
+        data_definition = DataDefinition()
+
+        data_definition.arrays.append(self._build_array(input_image, "DXYZ"))
```

Comment:
I've merge this PR, #238 , so now in-memory input image array asnumpy() now has index ordering of DHW, and the seg image also has this ordering. The numpy array of images coming from DICOM loader, and leaving the segmentation operator are now supposed to be similar to the numpy array from ITK.

Reply:
Thanks, @MMelQin. I updated the Pull Request and removed the workaround.